Functions/ImageProcessing/xASL_im_ASLSubtractionAveraging.m

```diff
@@ -0,0 +1,356 @@
+function [PWI, PWI3D, PWI4D, x] = xASL_im_ASLSubtractionAveraging(x, path_ASL4D, PWI4D, PWI3D)
+%xASL_im_ASLSubtractionAveraging Central function for ASL subtraction and averaging
+%
+% FORMAT: [PWI, PWI3D, PWI4D, x] = xASL_im_ASLSubtractionAveraging(x, ASL4D [, PWI4D, PWI3D])
+%
+% INPUT:
+%   x       - structure containing fields with all information required to run this function (REQUIRED)
+%             with the following information:
+%   x.modules.asl.bTimeEncoded - if we perform time encoding instead of
+%             pair-wise subtraction (boolean, REQUIRED)
+%             NB: see xASL_quant_HadamardDecoding for fields required by
+%             time encoding
+%   x.Q     - field containing quantification parameters (REQUIRED)
+%   x.Q.EchoTime - numerical vector. If this is a scalar, it will be extended to the number of image volumes (REQUIRED)
+%   x.Q.Initial_PLD - numerical vector. If this is a scalar, it will be extended to the number of image volumes (REQUIRED)
+%   x.Q.LabelingDuration - numerical vector. If this is a scalar, it will be extended to the number of image volumes (REQUIRED)
+% 
+%   path_ASL4D - Path to the encoded ASL4D image we want to decode (STRING OR 4D MATRIX, REQUIRED)
+%             Alternatively, this can contain the image matrix
+%             (xASL_io_Nifti2Im below allows both path and image matrix inputs).
+%             Best if this is a path, so any motion correction can be applied.
+%   PWI4D   - likewise, see explanation below. If this is provided, it will override the PWI4D calculation below (STRING OR 4D MATRIX, OPTIONAL,
+%               DEFAULT = generated below from ASL4D)
+%   PWI3D   - likewise, see explanation below. If this is provided, it will override the PWI4D calculation below (STRING OR 4D MATRIX, OPTIONAL,
+%               DEFAULT = generated below from PWI4D)
+%
+% OUTPUT:
+%   PWI     - 3D image matrix, see explanation below
+%   PWI3D   - 4D image matrix, see explanation below
+%   PWI4D   - 4D image matrix, see explanation below
+%   x       - updated structure, with the following quantification parameters:
+%             fitting to the nomenclature explained below 
+%
+%   x.Q.EchoTimeLabelingDuration_PWI4D
+%   x.Q.Initial_PLD_PWI4D
```

Comment:
You say Initial_PLD here, but use InitialPLD later - please check for consistency